prestashop1.6/controllers/admin/AdminTawktoController.php

```diff
@@ -150,68 +139,42 @@ public function ajaxProcessSetWidget()
             die(Tools::jsonEncode(array('success' => false)));
         }
 
-        $shopId = 1;
-        $shops = Shop::getShops();
-        $domain = addslashes(trim($_REQUEST['domain']));
-        if (count($shops) && !empty($domain)) {
-            foreach ($shops as $shop) {
-                if ($domain && $shop['domain'] == $domain) {
-                    $shopId = (int) $shop['id_shop'];
-                }
-            }
-        }
-
-        $pageKey = TawkTo::TAWKTO_WIDGET_PAGE_ID."_{$shopId}";
+        $pageKey = TawkTo::TAWKTO_WIDGET_PAGE_ID;
         Configuration::updateValue($pageKey, $pageId);
 
-        $widgetKey = TawkTo::TAWKTO_WIDGET_WIDGET_ID."_{$shopId}";
+        $widgetKey = TawkTo::TAWKTO_WIDGET_WIDGET_ID;
         Configuration::updateValue($widgetKey, $widgetId);
 
-        $userKey = TawkTo::TAWKTO_WIDGET_USER."_{$shopId}";
+        $userKey = TawkTo::TAWKTO_WIDGET_USER;
         Configuration::updateValue($userKey, $this->context->employee->id);
 
         die(Tools::jsonEncode(array('success' => true)));
     }
 
     public function ajaxProcessRemoveWidget()
     {
-        $shopId = 1;
-        $shops = Shop::getShops();
-        $domain = addslashes(trim($_REQUEST['domain']));
-        if (count($shops) && !empty($domain)) {
-            foreach ($shops as $shop) {
-                if ($domain && $shop['domain'] == $domain) {
-                    $shopId = (int) $shop['id_shop'];
-                }
+        $keys = array(
+            TawkTo::TAWKTO_WIDGET_PAGE_ID,
+            TawkTo::TAWKTO_WIDGET_WIDGET_ID,
+            TawkTo::TAWKTO_WIDGET_USER
+        );
+
+        foreach ($keys as $key) {
+            if (Shop::getContext() == Shop::CONTEXT_ALL) {
+                $this->deleteGlobalConfig($key);
+            } else {
+                // Configuration::deleteFromContext method cannot be used by
+                // 'All Shops' or the current shop context is 'CONTEXT_ALL'.
```

Comment:
By this, this is what I meant by `cannot be used by 'All Shops' or 'CONTEXT_ALL'`. The method returns immediately

https://github.com/PrestaShop/PrestaShop/blob/1.6.1.24/classes/Configuration.php#L500